### DISCUSSION TITLE
Docker container for arm64

### DISCUSSION THREAD

**Discussion by mibers** [labgrid-project/labgrid#1437] [Q&A]
Hi guys,

I wanna use labgrid dockerized on raspberry pi(s) but I only found amd64 based images on docker hub. So my questions:

- Is it planned to build images also arm65 based?
- How do the images on docker hub relate to the branches on GitHub, from which branches are they built?

For now I build them myself as described here:
https://github.com/labgrid-project/labgrid/tree/master/dockerfiles
and they work fine but just pulling most recent images from docker hub
might be more convenient.

Thx and best regards,
Martin I.

**Comment by jluebbe** [comment]
We don't use containers in our labs at Pengutronix and the existing ones are mainly a community effort. If you are interested in helping with maintenance or improvements, that would be great. :)

**Comment by Bastian-Krause** [comment]
> * Is it planned to build images also arm65 based?

#1307 introduces arm64 docker images.

> * How do the images on docker hub relate to the branches on GitHub, from which branches are they built?

The images are built from master and pushed, see https://github.com/labgrid-project/labgrid/actions/workflows/docker.yml

Note that running an exporter in a container requires passing through the host's udev, see https://github.com/labgrid-project/labgrid/tree/master/dockerfiles#labgrid-exporter-usage

**Reply by Bastian-Krause** [reply]
> > * Is it planned to build images also arm65 based?
> 
> #1307 introduces arm64 docker images.

The PR was merged. See https://hub.docker.com/u/labgrid for the arm64 variants.
